src/uwtools/drivers/driver.py

```diff
@@ -7,25 +7,35 @@
 
 from uwtools import config_validator
 from uwtools.logger import Logger
+from uwtools.scheduler import BatchScript
 
 
 class Driver(ABC):
     """
     An abstract class representing drivers for various NWP tools.
     """
 
-    def __init__(self, config_file: str, log: Optional[Logger] = None):
+    def __init__(
+        self,
+        config_file: str,
+        log: Optional[Logger] = None,
+        dry_run: bool = False,
+        outfile: Optional[str] = None,
```

Comment:
Is `outfile` a concept that would apply to **all** drivers? My guess is "no". If we need this option for the `FV3Forecast` subclass, it could move to that constructor of that class. Thoughts?

Reply:
Since I'm using 'outfile' here as the output batch script file name, it would apply to all drivers - I guess the better question is whether its an appropriate name for the sake of menu symmetry or if it should be a bool flag or something like "batch_file_name"?

Reply:
Do we (currently) need this value -- the path to the batch script to write -- anywhere but in the `run()` method? If no, could it (or, like you say, a more explicitly-named e.g. `batch_script_name`) be an argument to that method instead? Then one might, elsewhere, do something like
```
fv3fcst = FV3Forecast(...)
fv3fcst.run(..., batch_script_name="whatever")
```
Is it accurate that we should only need `batch_script_name` instead of `batch_script_path` because we'll be writing it to a run directory whose path is known by other means? Does the filename itself even need to be configurable? Could/should we have a standard name for a forecast batch script?

Reply:
True! I can move that over to run() specifically. For the name, hmm, 'can' and 'should' give different answers here - I'm inclined to give it just a name and pull `run_directory` from the YAML for the write?

Reply:
I like that. If we find a use case where users need to use a different name, we could field a feature request.